manifests/uid.pp

```diff
@@ -0,0 +1,20 @@
+#
+# Provides system-wide defaults for UID settings
+#
+# @param min
+#   The lowest allowed regular user UID for the system
+#
+# @param max
+#   The highest allowed regular user UID for the system
+#
+#   * If not defined, applications should simply do what makes sense for them
+#     internally
+#
+# @author SIMP Team - https://simp-project.com
+#
+class simp_options::uid (
+  Integer[0]           $min = pick(fact('login_defs.uid_min'), 1000),
```

Comment:
Why the pick()?  The new fact sets a default value for uid_min if it is not specified in/etc/login.defs.